File: pages/[locale]/clean.js
import Head from "next/head";
import { useRouter } from 'next/router';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHome } from "@fortawesome/free-solid-svg-icons";
import Image from "next/image";
import { useTranslation } from 'next-i18next';
import { getStaticPaths, makeStaticProps } from '../../lib/getStatic';

/**
 * Function that generates the necessary static paths and props manually
 * This is to fix an issue with next18 translations
 */
const getStaticProps = makeStaticProps('common');
export { getStaticPaths, getStaticProps };

/**
 * This page is displayed after a scan if no virus was found, aka the system is "clean"
 * Displays a title, an image, and a button to navigate back to the home page.
 * @returns A full HTML page
 */
export default function Clean() {
    const router = useRouter();
    const { t } = useTranslation('common');

    /**
     * Function to navigate back to the home page
     */
    const backHome = () => {
        router.push('/');
    }

    return (
        <>
            <Head>
                {/* Set the page title using translated text */}
                <title>{t('clean_title')}</title>
            </Head>
            <div className="flex items-center justify-center h-screen flex-col">
                <h1 className="text-center mb-10 pt-4 font-medium ltext-5xl text-maingreen">
                    {/* Display the translated title */}
                    {t('clean_title')}
                </h1>
                <Image 
                    src="/images/success_image.png" 
                    alt="Success" 
                    className="max-w-[30%]" 
                    width={500}
                    height={500}
                />
                <button onClick={backHome} 
                    type="button" 
                    className="inline-block px-6 py-2.5 m-10 bg-mainred text-white font-medium text-xs leading-tight uppercase rounded shadow-md">
                    <FontAwesomeIcon
                        icon={faHome}
                        size="1x"
                        className="pr-1"
                    />
                    {/* Display the translated button text */}
                    {t('back_btn')}
                </button>
            </div>
        </>
    )
}

File: src/pages/clean.rs
use crate::components::home_button::HomeButton;
use crate::i18n::use_i18n;
use leptos::*;
use leptos_i18n::t;

/// Clean page
/// The page that shows if no threat was found on the scanned target. It simply contains an image
/// that shows a success message and a button to go back to the home page.
#[component]
pub fn Clean() -> impl IntoView {
    let i18n = use_i18n();
    view! {
        <div class="h-screen">
            <div class="flex h-full justify-center p-12 text-center">
                <div class="flex justify-center items-center h-full">
                    <div class="w-full">
                        <h1 class="inline-block align-middle p-2 font-medium leading-tight text-5xl mt-0 mb-2 text-maingreen">
                            {t!(i18n, clean_title)}
                        </h1>
                        <div class="flex justify-center">
                            <img
                                src="/images/success_image.png"
                                alt="Success"
                                class="h-auto max-w-[60%]"
                                width="500"
                                height="500"
                            />
                        </div>
                        <HomeButton />
                    </div>
                </div>
            </div>
        </div>
    }
}
